### ISSUE TITLE
Awaiting a kernel handle - documentation error ?

### ISSUE THREAD

**Issue by mediabuff** [microsoft/cppwinrt#865]
https://docs.microsoft.com/en-us/windows/uwp/cpp-and-winrt-apis/concurrency-2

The documentation claims the following is a problem:
`
IAsyncAction Async(HANDLE event)
{
    co_await DoWorkAsync();
    co_await resume_on_signal(event); // The incoming handle is not valid here.
}
`

It's not true that when the HANDLE goes out of scope, the kernel 'CloseHandle's it. It a valid a handle still in the lambda and should work fine.

What is a problem definitely when we don't take a strong reference to 'winrt::handle event' as the winrt destructor willl 'CloseHandle'.
`
IAsyncAction Async(winrt::handle event)
{
    co_await DoWorkAsync();
    co_await resume_on_signal(event); // The incoming handle *is* valid here.
}
`

**Comment by kennykerr** [comment]
The point of the documentation is to highlight that this is unsafe. While the `HANDLE` *may* be valid, there's no way for the coroutine to ensure that it is valid since it does not own a reference to the HANDLE. It is therefore important that you either pass in an owning handle e.g. `winrt::handle` or use something like `DuplicateHandle` to ensure that the coroutine will not outlive the handle that it expects to wait on.
